xcodeproj/internal/xcschemes/xcschemes.bzl

```
@@ -0,0 +1,1125 @@
+            provided, and `None` or `{}` will result in no environment
+            variables.
+
+            Each value of the `dict` can either be a string or a value returned
+            by [`xcschemes.env_value`](#xcschemes.env_value). If a value is a
+            string, it will be transformed into `xcschemes.env_value(value)`.
+            For example,
+            ```
+            xcschemes.profile(
+                env = {
+                    "VAR1": "value 1",
+                    "VAR 2": xcschemes.env_value("value2", enabled = False),
+                },
+            )
+            ```
+            will be transformed into:
+            ```
+            xcschemes.profile(
+                env = {
+                    "VAR1": xcschemes.env_value("value 1"),
+                    "VAR 2": xcschemes.env_value("value2", enabled = False),
+                },
+            )
+            ```
+        env_include_defaults: Whether to include the rules_xcodeproj provided
+            default Bazel environment variables (e.g.
+            `BUILD_WORKING_DIRECTORY` and `BUILD_WORKSPACE_DIRECTORY`), in
+            addition to any set by [`env`](#xcschemes.profile-env).
+        launch_target: The target to launch when profiling.
+
+            Can be `None`, a label string, or a value returned by
+            [`xcschemes.launch_target`](#xcschemes.launch_target). If a label
+            string, `xcschemes.launch_target(label_str)` will be used. If
+            `None`, `xcschemes.launch_target()` will be used, which means no
+            launch target will be set (i.e. the `Executable` dropdown will be
+            set to `None`).
+        use_run_args_and_env: Whether the `Use the Run action's arguments
+            and environment variables` checkbox is checked.
+
+            If `True`, command-line arguments and environment variables will
+            still be set as defined by [`args`](#xcschemes.profile-args) and
+            [`env`](#xcschemes.profile-env), but will be ignored by Xcode unless
+            you manually uncheck this checkbox in the scheme. If `None`, `True`
+            will be used if [`args`](#xcschemes.profile-args) and
+            [`env`](#xcschemes.profile-env) are both `"inherit"`, otherwise
+            `False` will be used.
+        xcode_configuration: The name of the Xcode configuration to use to build
+            the targets referenced in the Profile action (i.e in the
+            [`build_targets`](#xcschemes.profile-build_targets) and
+            [`launch_target`](#xcschemes.profile-launch_target) attributes).
+
+            If not set, the value of
+            [`xcodeproj.default_xcode_configuration`](#xcodeproj-default_xcode_configuration)
+            is used.
+    """
+    if use_run_args_and_env == None:
+        use_run_args_and_env = args == "inherit" and env == "inherit"
+
+    return struct(
+        args = args or [],
+        build_targets = build_targets or [],
+        env = env or [],
```

`EMPTY_LIST` here and throughout this file as well?
```suggestion
        args = args or EMPTY_LIST,
        build_targets = build_targets or EMPTY_LIST,
        env = env or EMPTY_LIST,
```

---

Not this file, since these values won't be stored in a provider or depset. I only use `TRUE_ARG` and `FALSE_ARG` because they are magic values, while the `EMPTY_*` class of constants are for memory efficiency only.